src/connector/src/sink/file_sink/opendal_sink.rs

```diff
@@ -178,43 +196,67 @@ pub struct OpenDalSinkWriter {
 enum FileWriterEnum {
     ParquetFileWriter(AsyncArrowWriter<Compat<FuturesAsyncWriter>>),
 }
-
-#[async_trait]
-impl SinkWriter for OpenDalSinkWriter {
-    async fn write_batch(&mut self, chunk: StreamChunk) -> Result<()> {
-        // Note: epoch is used to name the output files.
-        // Todo: after enabling sink decouple, use the new naming convention.
-        let epoch = self.epoch.ok_or_else(|| {
-            SinkError::File("epoch has not been initialize, call `begin_epoch`".to_string())
-        })?;
+impl OpenDalSinkWriter {
```

Comment:
We have plenty of good documentations for the newly added methods, which is super helpful for understanding the code logics. But we can do even better if we can organize the codes in a cleaner structure. Currently the logics of batching, writing chunks and committing (close writer) are mixed in some of the methods. I suggest we clearly separate them.


Example:
```rust
impl OpenDalSinkWriter {
    // Methods to write chunks only.
    pub async fn write_batch(...) -> ... {
       ...
       writer.write(...)
       // Update current_bached_row_num
       ...
    }

    // Methods to commit only
    pub async fn commit(...) -> ... {
        if writer.bytes_written() > 0 {
            writer.close().await?;
            ...
        }
    }

    // Methods to whether batching is finished
    pub fn should_commit(...) -> bool {
        self.duration_seconds_since_writer_created() > ... || self.current_bached_row_num > ...
    }

    
    pub fn try_commit(...) -> ... {
        if should_commit(...) {
            self.commit(...)
        }
        ...
    }
}

```

BatchingLogSinker only needs to use write_batch and try_commit.

Reply:
Reorganize as follow, put all those write methods together
```
/// Method for writing files into file system.
///
/// # Write Chunk Methods
/// - `write_batch_and_try_finish`: Writes a stream chunk to the sink and tries to close the writer according to the batching strategy.
///
/// # Close Writer Methods
/// - `try_finish`: Attempts to complete the file write using the writer's internal batching strategy.
/// - `should_finish`: Force close writer regardless of whether it trigger batching strategy.
///
/// # Batching Condition Methods
/// - `try_finish_write_via_batched_rows`: Checks if writing can be completed based on the number of batched rows.
/// - `try_finish_write_via_rollover_interval`: Checks if writing can be completed based on the rollover time interval.
```